pkg/dynamiclistener/server.go

```diff
@@ -130,20 +130,9 @@ func (s *Server) updateIPs(savedIPs map[string]bool) map[string]bool {
 		return savedIPs
 	}
 
-	certs := map[string]string{}
-	s.Lock()
-	for key, cert := range s.certs {
-		certs[key] = certToString(cert)
-	}
-	s.Unlock()
-
-	if !reflect.DeepEqual(certs, cfg.GeneratedCerts) {
-		cfg = cfg.DeepCopy()
-		cfg.GeneratedCerts = certs
-		cfg, err = s.listenConfigStorage.Update(cfg)
-		if err != nil {
-			return savedIPs
-		}
+	err = s.reconcileGeneratedCert(cfg)
+	if err != nil {
+		return savedIPs
```

Comment:
This is done only as a backup, in case the original save fails. If we want to simplify we can just do the save on generation below.